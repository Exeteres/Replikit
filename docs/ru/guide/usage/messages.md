# Сообщения

Пакет `@replikit/messages` содержит весь необходимый инструментарий для работы с сообщениями.

## MessageBuilder

Самый мощный интерфейс в Replikit для создания сообщений; использует паттерн `Fluent Builder`.

```ts
import { MessageBuilder } from "@replikit/messages";
import { AttachmentType } from "@replikit/core";

// Простое сообщение с текстом
const message1 = new MessageBuilder()
    .addText("Hello")
    .build();

// Сообщение, состоящее из нескольких строк
const message2 = new MessageBuilder()
    .addLine("one")
    .addLine("two")
    .addLines(["three", "four"])
    .build();

// Сообщение, содержащее вложение
const message3 = new MessageBuilder()
    .addAttachmentByUrl(AttachmentType.Photo, "https://example.com/photo.png")
    .build();
```

## Сокращения

Для создания простых сообщений существуют функции - сокращения:

```ts
import { fromText, fromCode, fromAttachment } from "@replikit/messages";
import { AttachmentType } from "@replikit/core";

// Простое сообщение с текстом
const message1 = fromText("Hello");

// Простое сообщение с кодом
const message2 = fromCode("Hello");

// Сообщение, содержащее вложение
const message3 = fromAttachment({
    type: AttachmentType.Photo,
    url: "https://example.com/photo.png"
});
```
