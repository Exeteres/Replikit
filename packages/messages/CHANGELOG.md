# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.22.1](https://github.com/Exeteres/Replikit/compare/v0.22.0...v0.22.1) (2020-10-23)

**Note:** Version bump only for package @replikit/messages





# [0.22.0](https://github.com/Exeteres/Replikit/compare/v0.21.0...v0.22.0) (2020-10-23)

**Note:** Version bump only for package @replikit/messages





# [0.21.0](https://github.com/Exeteres/Replikit/compare/v0.20.0...v0.21.0) (2020-10-17)

**Note:** Version bump only for package @replikit/messages





# [0.20.0](https://github.com/Exeteres/Replikit/compare/v0.19.0...v0.20.0) (2020-09-28)


### Features

* Add ability to create multiple button rows ([2a83a74](https://github.com/Exeteres/Replikit/commit/2a83a7432c102ff77e71cd4d542fdb4ec99ac7ad))





# [0.19.0](https://github.com/Exeteres/Replikit/compare/v0.18.2...v0.19.0) (2020-09-18)


### Features

* **messages:** Add addButton method ([1d7f3a0](https://github.com/Exeteres/Replikit/commit/1d7f3a0adb2b31e02707f4093e0ec7d828797403))
* **messages:** Add OutMessageLike type and utils ([a9fb59c](https://github.com/Exeteres/Replikit/commit/a9fb59cb432419fadd3a51e79d135ac29c9087b5))
* **messages:** Add token factories and support for buttons ([2275483](https://github.com/Exeteres/Replikit/commit/227548389a70cf64756c298951241ea96fbd77c4))





## [0.18.2](https://github.com/Exeteres/Replikit/compare/v0.18.1...v0.18.2) (2020-09-09)

**Note:** Version bump only for package @replikit/messages





## [0.18.1](https://github.com/Exeteres/Replikit/compare/v0.18.0...v0.18.1) (2020-09-09)

**Note:** Version bump only for package @replikit/messages





# [0.18.0](https://github.com/Exeteres/Replikit/compare/v0.17.3...v0.18.0) (2020-09-09)

**Note:** Version bump only for package @replikit/messages





## [0.17.3](https://github.com/Exeteres/Replikit/compare/v0.17.2...v0.17.3) (2020-08-25)

**Note:** Version bump only for package @replikit/messages





## [0.17.2](https://github.com/Exeteres/Replikit/compare/v0.17.1...v0.17.2) (2020-08-24)

**Note:** Version bump only for package @replikit/messages





## [0.17.1](https://github.com/Exeteres/Replikit/compare/v0.17.0...v0.17.1) (2020-08-21)

**Note:** Version bump only for package @replikit/messages





# [0.17.0](https://github.com/Exeteres/Replikit/compare/v0.16.0...v0.17.0) (2020-08-21)

**Note:** Version bump only for package @replikit/messages





# [0.16.0](https://github.com/Exeteres/Replikit/compare/v0.15.1...v0.16.0) (2020-08-18)


### Features

* **messages:** Add methods to work with message headers ([f412a28](https://github.com/Exeteres/Replikit/commit/f412a280c4a185a8066fec2b8492bc4afd8d8431))





## [0.15.1](https://github.com/Exeteres/Replikit/compare/v0.15.0...v0.15.1) (2020-08-13)

**Note:** Version bump only for package @replikit/messages





# [0.15.0](https://github.com/Exeteres/Replikit/compare/v0.14.0...v0.15.0) (2020-08-13)

**Note:** Version bump only for package @replikit/messages





# [0.14.0](https://github.com/Exeteres/Replikit/compare/v0.13.0...v0.14.0) (2020-08-05)


### Features

* **messages:** Add ability to use addLine method without text ([3886f11](https://github.com/Exeteres/Replikit/commit/3886f113aa94943afea2bd5f4f94462eccec055e))





# [0.13.0](https://github.com/Exeteres/Replikit/compare/v0.12.1...v0.13.0) (2020-08-04)


### Features

* **messages:** Add ability to use pipe method in MessageBuilder ([f5fe039](https://github.com/Exeteres/Replikit/commit/f5fe0391708753b7404e6c695a77cd793e83120c))
* **messages:** Add alias to use message with metadata as metadata ([e315a25](https://github.com/Exeteres/Replikit/commit/e315a256dfeef85137f640e08176d14c84a1638f))





# [0.12.0](https://github.com/Exeteres/Replikit/compare/v0.11.2...v0.12.0) (2020-07-30)

**Note:** Version bump only for package @replikit/messages





## [0.11.2](https://github.com/Exeteres/Replikit/compare/v0.11.1...v0.11.2) (2020-07-24)

**Note:** Version bump only for package @replikit/messages





## [0.11.1](https://github.com/Exeteres/Replikit/compare/v0.11.0...v0.11.1) (2020-07-24)

**Note:** Version bump only for package @replikit/messages






# [0.11.0](https://github.com/Exeteres/Replikit/compare/v0.10.1...v0.11.0) (2020-07-24)


### Features

* **messages:** Add MessageLike type ([046ca1b](https://github.com/Exeteres/Replikit/commit/046ca1b8a19308790f3e0a424a0cb9f3cb7dc7dd))





## [0.10.1](https://github.com/Exeteres/Replikit/compare/v0.10.0...v0.10.1) (2020-06-30)

**Note:** Version bump only for package @replikit/messages






# [0.10.0](https://github.com/Exeteres/Replikit/compare/v0.9.0...v0.10.0) (2020-06-29)

**Note:** Version bump only for package @replikit/messages





# [0.9.0](https://github.com/Exeteres/Replikit/compare/v0.8.0...v0.9.0) (2020-06-28)


### Features

* Add advanced metadata processing ([54c7820](https://github.com/Exeteres/Replikit/commit/54c782021f5ebc16d784f210d273f8d7490d2af5))





# [0.8.0](https://github.com/Exeteres/Replikit/compare/v0.7.2...v0.8.0) (2020-06-27)


### Features

* **messages:** Add MessageBuilder methods for working with replies ([015e3f2](https://github.com/Exeteres/Replikit/commit/015e3f23b3f166b46aecb2b98f6f447661b8576d))





## [0.7.2](https://github.com/Exeteres/Replikit/compare/v0.7.1...v0.7.2) (2020-06-23)

**Note:** Version bump only for package @replikit/messages





# [0.7.0](https://github.com/Exeteres/Replikit/compare/v0.6.0...v0.7.0) (2020-06-23)

**Note:** Version bump only for package @replikit/messages





# [0.6.0](https://github.com/Exeteres/Replikit/compare/v0.5.3...v0.6.0) (2020-06-21)

**Note:** Version bump only for package @replikit/messages





## [0.5.3](https://github.com/Exeteres/Replikit/compare/v0.5.2...v0.5.3) (2020-06-20)

**Note:** Version bump only for package @replikit/messages





## [0.5.2](https://github.com/Exeteres/Replikit/compare/v0.5.1...v0.5.2) (2020-06-20)

**Note:** Version bump only for package @replikit/messages





# 0.5.0 (2020-06-20)

**Note:** Version bump only for package @replikit/messages






# [0.4.0](https://github.com/Exeteres/Replikit/compare/v0.3.0...v0.4.0) (2020-06-19)

**Note:** Version bump only for package @replikit/messages





# [0.3.0](https://github.com/Exeteres/Replikit/compare/v0.2.0...v0.3.0) (2020-06-14)

**Note:** Version bump only for package @replikit/messages





# [0.2.0](https://github.com/Exeteres/Replikit/compare/v0.1.0...v0.2.0) (2020-06-13)

**Note:** Version bump only for package @replikit/messages
