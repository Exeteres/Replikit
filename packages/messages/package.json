{
    "name": "@replikit/messages",
    "version": "0.22.1",
    "license": "MIT",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "typings"
    ],
    "scripts": {
        "build": "tsc -p tsconfig.build.json"
    },
    "dependencies": {
        "@replikit/core": "^0.22.1"
    }
}
