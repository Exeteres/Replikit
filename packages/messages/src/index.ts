export * from "./shortcuts";
export * from "./utils";
export * from "./messageBuilder";
