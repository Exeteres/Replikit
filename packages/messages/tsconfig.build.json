{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/messages": ["messages/src"]
        }
    },
    "include": ["src"]
}
