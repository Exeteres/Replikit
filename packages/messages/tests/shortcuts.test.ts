import { fromText, fromCode } from "@replikit/messages";

describe("shortcuts", () => {
    it("should create a message with text", () => {
        const message = fromText("test");
        expect(message).toMatchSnapshot();
    });

    it("should create a message with code", () => {
        const message = fromCode("test");
        expect(message).toMatchSnapshot();
    });
});
