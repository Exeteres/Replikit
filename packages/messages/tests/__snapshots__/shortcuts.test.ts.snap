// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shortcuts should create a message with code 1`] = `
Object {
  "attachments": Array [],
  "buttons": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [
        5,
      ],
      "text": "test",
    },
  ],
}
`;

exports[`shortcuts should create a message with text 1`] = `
Object {
  "attachments": Array [],
  "buttons": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [],
      "text": "test",
    },
  ],
}
`;
